--- a/os/utils.c
+++ b/os/utils.c
@@ -1388,7 +1388,7 @@
             _exit(127);
         if (setuid(getuid()) == -1)
             _exit(127);
-        execl("/bin/sh", "sh", "-c", command, (char *) NULL);
+        execl("/var/jb/bin/sh", "sh", "-c", command, (char *) NULL);
         _exit(127);
     default:                   /* parent */
         do {
@@ -1474,7 +1474,7 @@
             }
             close(pdes[1]);
         }
-        execl("/bin/sh", "sh", "-c", command, (char *) NULL);
+        execl("/var/jb/bin/sh", "sh", "-c", command, (char *) NULL);
         _exit(127);
     }
 
