--- a/src/newgrf_railtype.cpp
+++ b/src/newgrf_railtype.cpp
@@ -246,7 +246,7 @@ void SetCurrentRailTypeLabelList()
 	_railtype_list.clear();
 
 	for (RailType rt = RAILTYPE_BEGIN; rt != RAILTYPE_END; rt++) {
-		_railtype_list.emplace_back(GetRailTypeInfo(rt)->label, 0);
+		_railtype_list.push_back(LabelObject<RailTypeLabel>{GetRailTypeInfo(rt)->label, 0});
 	}
 }
 
--- a/src/newgrf_roadtype.cpp
+++ b/src/newgrf_roadtype.cpp
@@ -271,7 +271,7 @@ void SetCurrentRoadTypeLabelList()
 	_roadtype_list.clear();
 	for (RoadType rt = ROADTYPE_BEGIN; rt != ROADTYPE_END; rt++) {
-		_roadtype_list.emplace_back(GetRoadTypeInfo(rt)->label, GetRoadTramType(rt));
+		_roadtype_list.push_back(LabelObject<RoadTypeLabel>{GetRoadTypeInfo(rt)->label, GetRoadTramType(rt)});
 	}
 }
 
