--- a/src/textfile_gui.cpp
+++ b/src/textfile_gui.cpp
@@ -232,7 +232,7 @@ void TextfileWindow::FindHyperlinksInMarkdown(Line &line, size_t line_index)
 	while (matcher != std::sregex_iterator()) {
 		std::smatch match = *matcher;
 
-		Hyperlink &link = this->links.emplace_back(line_index, 0, 0, match[2].str());
+		Hyperlink &link = this->links.emplace_back(Hyperlink{line_index, 0, 0, match[2].str()});
 
 		HyperlinkType link_type = ClassifyHyperlink(link.destination, this->trusted);
 		StringControlCode link_colour;
@@ -335,7 +335,7 @@ void TextfileWindow::AppendHistory(const std::string &filepath)
 {
 	this->history.erase(this->history.begin() + this->history_pos + 1, this->history.end());
 	this->UpdateHistoryScrollpos();
-	this->history.emplace_back(filepath, 0);
+	this->history.emplace_back(HistoryEntry{filepath, 0});
 	this->EnableWidget(WID_TF_NAVBACK);
 	this->DisableWidget(WID_TF_NAVFORWARD);
 	this->history_pos = this->history.size() - 1;
@@ -496,7 +496,7 @@ void TextfileWindow::AfterLoadText()
 		if (!line.text.empty() && line.text[0] == '#') {
 			this->jumplist.push_back(line_index);
 			this->lines[line_index].colour = TC_GOLD;
-			this->link_anchors.emplace_back(line_index, 0, 0, MakeAnchorSlug(line.text));
+			this->link_anchors.emplace_back(Hyperlink{line_index, 0, 0, MakeAnchorSlug(line.text)});
 		}
 	}
 }
@@ -891,7 +891,7 @@ void TextfileWindow::LoadTextfile(const std::string &textfile, Subdirectory subdi
 	this->filepath = textfile;
 	this->filename = this->filepath.substr(this->filepath.find_last_of(PATHSEP) + 1);
 	/* If it's the first file being loaded, add to history. */
-	if (this->history.empty()) this->history.emplace_back(this->filepath, 0);
+	if (this->history.empty()) this->history.emplace_back(HistoryEntry{this->filepath, 0});
 
 	/* Process the loaded text into lines, and do any further parsing needed. */
 	this->LoadText(sv_buf);
