--- a/src/strings.cpp
+++ b/src/strings.cpp
@@ -2144,7 +2144,7 @@
 
 /* Win32 implementation in win32.cpp.
  * OS X implementation in os/macosx/macos.mm. */
-#if !(defined(_WIN32) || defined(__APPLE__))
+#if !(defined(_WIN32) || defined(__APPLE__)) || defined(OPENTTD_XIOS)
 /**
  * Determine the current charset based on the environment
  * First check some default values, after this one we passed ourselves
