--- a/src/file/SDL_rwops.c
+++ b/src/file/SDL_rwops.c
@@ -50,7 +50,7 @@
 #include "SDL_endian.h"
 #include "SDL_rwops.h"

-#ifdef __APPLE__
+#if defined(__APPLE__) && !defined(SDL_XIOS_WAYLAND)
 #include "cocoa/SDL_rwopsbundlesupport.h"
 #endif /* __APPLE__ */

@@ -626,7 +626,7 @@
     rwops->type = SDL_RWOPS_WINFILE;
 #elif defined(HAVE_STDIO_H)
     {
-#if defined(__APPLE__) && !defined(SDL_FILE_DISABLED) // TODO: add dummy?
+#if defined(__APPLE__) && !defined(SDL_FILE_DISABLED) && !defined(SDL_XIOS_WAYLAND) // TODO: add dummy?
         FILE *fp = SDL_OpenFPFromBundleOrFallback(file, mode);
 #elif defined(__WINRT__)
         FILE *fp = NULL;
