--- a/external/nss/UnpackedTarball_nss.mk
+++ b/external/nss/UnpackedTarball_nss.mk
@@ -36,4 +36,5 @@ $(eval $(call gb_UnpackedTarball_add_patches,nss,\
     $(if $(filter iOS,$(OS)), \
-        external/nss/nss-ios.patch) \
+        external/nss/nss-ios.patch \
+        external/nss/nss-ios-static-builtins.patch.0) \
     $(if $(filter ANDROID,$(OS)), \
         external/nss/nss-android.patch.1) \
--- a/external/expat/StaticLibrary_expat.mk
+++ b/external/expat/StaticLibrary_expat.mk
@@ -50 +50 @@
-else ifeq ($(OS),MACOSX)
+else ifeq ($(OS),$(filter MACOSX iOS,$(OS)))
